src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp

```diff
@@ -136,3 +136,206 @@ std::vector<Profile> WslDistroGenerator::GenerateProfiles()
 
     return profiles;
 }
+
+// Function Description:
+// - Create a list of Profiles for each distro listed in names.
+// - Skips distros that are utility distros for docker (see GH#3556)
+// Arguments:
+// - names: a list of distro names to turn into profiles
+// Return Value:
+// - the list of profiles we've generated.
+std::vector<Profile> _namesToProfiles(const std::vector<std::wstring>& names)
+{
+    std::vector<Profile> profiles;
+    for (const auto& distName : names)
+    {
+        if (distName.substr(0, std::min(distName.size(), DockerDistributionPrefix.size())) == DockerDistributionPrefix)
+        {
+            // Docker for Windows creates some utility distributions to handle Docker commands.
+            // Pursuant to GH#3556, because they are _not_ user-facing we want to hide them.
+            continue;
+        }
+
+        auto WSLDistro{ CreateDefaultProfile(distName) };
+
+        WSLDistro.Commandline(L"wsl.exe -d " + distName);
+        WSLDistro.DefaultAppearance().ColorSchemeName(L"Campbell");
+        WSLDistro.StartingDirectory(DEFAULT_STARTING_DIRECTORY);
+        WSLDistro.Icon(L"ms-appx:///ProfileIcons/{9acb9455-ca41-5af7-950f-6bca1bc9722f}.png");
+        profiles.emplace_back(WSLDistro);
+    }
+    return profiles;
+}
+
+// The WSL entries are structured as such:
+// HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+//   ⌞ {distroGuid}
+//     ⌞ DistributionName: {the name}
+const wchar_t RegKeyLxss[] = L"Software\\Microsoft\\Windows\\CurrentVersion\\Lxss";
+const wchar_t RegKeyDistroName[] = L"DistributionName";
+
+// Function Description:
+// - Open the reg key the root of the WSL data, in HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss
+// Arguments:
+// - <none>
+// Return Value:
+// - the HKEY if it exists and we can read it, else nullptr
+HKEY _openWslRegKey()

```

Comment:
make these static :smile:

Reply:
also, we may want to use the `wil` helpers here. `wil::unique_hkey` etc

Reply:
oh my god there's a wil registry_watcher

_like, we could reload the settings if the lxss key changes_

not saying that we should do this now, but... whoa